Unable to set license header styles

### Description
I am currently working on a project where I am mixing some Java and some Kotlin source files, every single one of them without a license header.
At first I attempted to run the `checkLicense` task with the following configuration:
```gradle
license {
    header = project.file('COPYING')

    include '**/*.java'
    include '**/*.kt'

    ext { /* Unnecessary */ }

    ignoreFailures = false

    style {
        java = 'BLOCK_COMMENT'
        kt = 'BLOCK_COMMENT'
    }
}
```
The build succeeded even if no one of the files present had the correct header. On the other hand, the plugin complained for every single file, both Java and Kotlin, that it could not find a matching header format.
After this, I tried finding out the source of the problem and, with some commenting, the build successfully failed when I commented out the entire `style` block. It seems like providing custom license header styles makes the plugin not only forget the defaults, but also not apply the specified styles.
I attached logs of both the run with the `style` block and the one without it.
In case you want to have a look at the setup of the entire project, it is available at [TheSilkMiner/SKL-Java-Interpreter@0.3 (commit 8b4063f)](https://github.com/TheSilkMiner/SKL-Java-Interpreter/tree/8b4063f06c4ff4169fd6f32cfa959e11966808bb).

### Logs
<details>
<summary><strong>Output of <code>./gradlew checkLicense --info --stacktrace</code> with <code>style</code> block</strong></summary>

```text
Initialized native services in: C:\Users\thesi\.gradle\native
Removing 0 daemon stop events from registry
Starting a Gradle Daemon (subsequent builds will be faster)
Starting process 'Gradle build daemon'. Working directory: C:\Users\thesi\.gradle\daemon\4.9 Command: C:\Program Files\Java\jdk1.8.0_171\bin\java.exe -XX:+HeapDumpOnOutOfMemoryError -Xmx1024m -Dfile.encoding=windows-1252 -Duser.country=US -Duser.language=en -Duser.variant -cp C:\Users\thesi\.gradle\wrapper\dists\gradle-4.9-all\491wbe0x5d54n9cojs2p0zv90\gradle-4.9\lib\gradle-launcher-4.9.jar org.gradle.launcher.daemon.bootstrap.GradleDaemon 4.9
Successfully started process 'Gradle build daemon'
An attempt to start the daemon took 7.242 secs.
The client will now receive all logging from the daemon (pid: 10656). The daemon log file: C:\Users\thesi\.gradle\daemon\4.9\daemon-10656.out.log
Starting build in new daemon [memory: 954.7 MB]
Using 2 worker leases.
Starting Build
Settings evaluated using settings file 'E:\GitHub\SKL-Java-Interpreter\settings.gradle'.
Projects loaded. Root project using build file 'E:\GitHub\SKL-Java-Interpreter\build.gradle'.
Included projects: [root project 'SKL-Java-Interpreter']
Parallel execution is an incubating feature.

> Configure project :
Evaluating root project 'SKL-Java-Interpreter' using build file 'E:\GitHub\SKL-Java-Interpreter\build.gradle'.
Compiling build file 'E:\GitHub\SKL-Java-Interpreter\build.gradle' using SubsetScriptTransformer.
Compiling build file 'E:\GitHub\SKL-Java-Interpreter\build.gradle' using BuildScriptTransformer.
All projects evaluated.
Selected primary task 'checkLicenses' from project :
Tasks to be executed: [task ':checkLicenseMain', task ':checkLicenseTest', task ':checkLicenses']
:checkLicenseMain (Thread[Daemon worker,5,main]) started.

> Task :checkLicenseMain
Task ':checkLicenseMain' is not up-to-date because:
  Task has not declared any outputs despite executing actions.
No matching header format found for src\main\java\net\thesilkminer\skl\interpreter\api\skd\exceptions\IllegalDatabaseSyntaxException.java
No matching header format found for src\main\java\net\thesilkminer\skl\interpreter\api\skd\holder\IDatabaseHolder.java
=== SNIPPED EQUAL LINES ===
No matching header format found for src\main\kotlin\net\thesilkminer\skl\interpreter\skd\tools\parser\Utilities.kt
No matching header format found for src\main\kotlin\net\thesilkminer\skl\interpreter\skd\tools\Support.kt
:checkLicenseMain (Thread[Daemon worker,5,main]) completed. Took 1.942 secs.
:checkLicenseTest (Thread[Daemon worker,5,main]) started.

> Task :checkLicenseTest
Task ':checkLicenseTest' is not up-to-date because:
  Task has not declared any outputs despite executing actions.
No matching header format found for src\test\java\org\apache\log4j\custom\MultipleFilesAppender.java
No matching header format found for src\test\kotlin\net\thesilkminer\skl\interpreter\skd\structure\ArrayEntityBaseTest.kt
=== SNIPPED EQUAL LINES ===
No matching header format found for src\test\kotlin\net\thesilkminer\skl\interpreter\skd\structure\PropertyBaseTest.kt
No matching header format found for src\test\kotlin\net\thesilkminer\skl\interpreter\skd\tools\parser\DeclaratorTest.kt
:checkLicenseTest (Thread[Daemon worker,5,main]) completed. Took 0.045 secs.
:checkLicenses (Thread[Task worker for ':',5,main]) started.

> Task :checkLicenses
Skipping task ':checkLicenses' as it has no actions.
:checkLicenses (Thread[Task worker for ':',5,main]) completed. Took 0.001 secs.

BUILD SUCCESSFUL in 39s
2 actionable tasks: 2 executed
```
</details>

<details>
<summary><strong>Output of <code>./gradlew checkLicense --info --stacktrace</code> without <code>style</code> block</strong></summary>

```text
Initialized native services in: C:\Users\thesi\.gradle\native
The client will now receive all logging from the daemon (pid: 10656). The daemon log file: C:\Users\thesi\.gradle\daemon\4.9\daemon-10656.out.log
Starting 3rd build in daemon [uptime: 2 mins 49.212 secs, performance: 99%, no major garbage collections]
Using 2 worker leases.
Starting Build
Settings evaluated using settings file 'E:\GitHub\SKL-Java-Interpreter\settings.gradle'.
Projects loaded. Root project using build file 'E:\GitHub\SKL-Java-Interpreter\build.gradle'.
Included projects: [root project 'SKL-Java-Interpreter']
Parallel execution is an incubating feature.

> Configure project :
Evaluating root project 'SKL-Java-Interpreter' using build file 'E:\GitHub\SKL-Java-Interpreter\build.gradle'.
All projects evaluated.
Selected primary task 'checkLicenses' from project :
Tasks to be executed: [task ':checkLicenseMain', task ':checkLicenseTest', task ':checkLicenses']
:checkLicenseMain (Thread[Daemon worker Thread 2,5,main]) started.

> Task :checkLicenseMain FAILED
Task ':checkLicenseMain' is not up-to-date because:
  Task has not declared any outputs despite executing actions.
:checkLicenseMain (Thread[Daemon worker Thread 2,5,main]) completed. Took 0.516 secs.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':checkLicenseMain'.
> License violations were found: src\main\java\net\thesilkminer\skl\interpreter\api\skd\exceptions\IllegalDatabaseSyntaxException.java, src\main\java\net\thesilkminer\skl\interpreter\api\skd\holder\IDatabaseHolder.java, **SNIP** src\main\kotlin\net\thesilkminer\skl\interpreter\skd\tools\parser\Utilities.kt, src\main\kotlin\net\thesilkminer\skl\interpreter\skd\tools\Support.kt

* Try:
Run with --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':checkLicenseMain'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:110)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:77)
=== SNIPPED INTERNAL GRADLE CALLS ===
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: net.minecrell.gradle.licenser.LicenseViolationException: License violations were found: src\main\java\net\thesilkminer\skl\interpreter\api\skd\exceptions\IllegalDatabaseSyntaxException.java, src\main\java\net\thesilkminer\skl\interpreter\api\skd\holder\IDatabaseHolder.java, **SNIP** src\main\kotlin\net\thesilkminer\skl\interpreter\skd\tools\parser\Utilities.kt, src\main\kotlin\net\thesilkminer\skl\interpreter\skd\tools\Support.kt
        at net.minecrell.gradle.licenser.tasks.LicenseCheck.checkFiles(LicenseCheck.groovy:73)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:46)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:786)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:753)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:131)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:120)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:99)
        ... 111 more


* Get more help at https://help.gradle.org

BUILD FAILED in 6s
1 actionable task: 1 executed
```
</details>

### Platform details
**Licenser version:** 0.4.1, added through `plugins` block
**Gradle version:** 4.9
**Groovy version:** 2.4.12
**Java version:** JDK 1.8.0u171 64-bit
**OS:** Windows 10 Home 64-bit (17735.rs5_release.180807-1451)
<details><summary><strong>Gradle daemon properties:</strong></summary>

```properties
# Gradle daemon related properties
org.gradle.warning.mode=all
org.gradle.parallel=true
org.gradle.daemon=true
org.gradle.console=auto
org.gradle.caching.debug=true
```
</details>

### Probable resolution hypothesis (based on commit e3a1b379a7223862bca96fc4b4aaa6501ff50a4b)
**Note: the following assumptions are solely based on speculations and are not tested.**
The `license` block in-code counterpart is the `LicenseExtension.groovy` file. There, [on line 76](https://github.com/Minecrell/licenser/blob/e3a1b379a7223862bca96fc4b4aaa6501ff50a4b/src/main/groovy/net/minecrell/gradle/licenser/LicenseExtension.groovy#L76), the `style` property is specified as an instance of `HeaderFormatRegistry`, i.e. a child of `CaseInsensitiveMap` with a more specific handling for `HeaderFormat`s.
The `style` property may thus be initialized as a "normal" `CaseInsensitiveMap<String>`. Its contents can then be used in the task to first create a `HeaderFormatRegistry` with the default `HeaderStyle`s and then override the few specified in the `style` property.

--- a/build.gradle
+++ b/build.gradle
@@ -14,8 +14,31 @@ group = 'org.cadixdev.gradle'
 version = '0.5.0'
 description = 'A simple license header manager for Gradle'
 
+repositories {
+    mavenCentral()
+}
+
 dependencies {
     implementation gradleApi()
+
+    testImplementation 'org.spockframework:spock-core:1.3-groovy-2.5'
+}
+
+sourceSets {
+    functionalTest {
+    }
+}
+gradlePlugin.testSourceSets(sourceSets.functionalTest)
+configurations.functionalTestImplementation.extendsFrom(configurations.testImplementation)
+
+tasks.register('functionalTest', Test) {
+    testClassesDirs = sourceSets.functionalTest.output.classesDirs
+    classpath = sourceSets.functionalTest.runtimeClasspath
+}
+
+tasks.named('check') {
+    // Run the functional tests as part of `check`
+    dependsOn(tasks.functionalTest)
 }
 
 license {

--- a/build.gradle
+++ b/build.gradle
@@ -18,27 +18,40 @@ repositories {
     mavenCentral()
 }
 
-dependencies {
-    implementation gradleApi()
-
-    testImplementation 'org.spockframework:spock-core:1.3-groovy-2.5'
-}
-
 sourceSets {
     functionalTest {
     }
 }
+
+configurations {
+    functionalTestImplementation.extendsFrom(testImplementation)
+    functionalTestRuntimeOnly.extendsFrom(testRuntimeOnly)
+}
 gradlePlugin.testSourceSets(sourceSets.functionalTest)
-configurations.functionalTestImplementation.extendsFrom(configurations.testImplementation)
 
 tasks.register('functionalTest', Test) {
+    description = 'Runs the functional tests.'
+    group = 'verification'
+
     testClassesDirs = sourceSets.functionalTest.output.classesDirs
     classpath = sourceSets.functionalTest.runtimeClasspath
+
+    mustRunAfter test
+}
+
+dependencies {
+    implementation gradleApi()
+
+    testImplementation 'org.spockframework:spock-core:1.3-groovy-2.5'
+    functionalTestImplementation gradleTestKit()
+}
+
+test {
+    useJUnit()
 }
 
 tasks.named('check') {
-    // Run the functional tests as part of `check`
-    dependsOn(tasks.functionalTest)
+    dependsOn(functionalTest)
 }
 
 license {

--- a/src/functionalTest/groovy/org/cadixdev/gradle/licenser/LicenserPluginFunctionalTest.groovy
+++ b/src/functionalTest/groovy/org/cadixdev/gradle/licenser/LicenserPluginFunctionalTest.groovy
@@ -1,0 +1,156 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015, Minecrell <https://github.com/Minecrell>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.cadixdev.gradle.licenser
+
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+import spock.lang.Specification
+
+class LicenserPluginFunctionalTest extends Specification {
+    @Rule
+    TemporaryFolder temporaryFolder = new TemporaryFolder()
+
+    def "can run licenseCheck task"() {
+        given:
+        def projectDir = temporaryFolder.newFolder()
+        new File(projectDir, "settings.gradle") << ""
+        new File(projectDir, "build.gradle") << """
+            plugins {
+                id('org.cadixdev.licenser')
+            }
+        """.stripIndent()
+
+        when:
+        def runner = GradleRunner.create()
+        runner.forwardOutput()
+        runner.withPluginClasspath()
+        runner.withArguments("licenseCheck")
+        runner.withProjectDir(projectDir)
+        def result = runner.build()
+
+        then:
+        result.task(":checkLicenses").outcome == TaskOutcome.UP_TO_DATE
+        result.task(":licenseCheck").outcome == TaskOutcome.UP_TO_DATE
+    }
+
+    def "can run licenseFormat task"() {
+        given:
+        def projectDir = temporaryFolder.newFolder()
+        new File(projectDir, "settings.gradle") << ""
+        new File(projectDir, "build.gradle") << """
+            plugins {
+                id('org.cadixdev.licenser')
+            }
+        """.stripIndent()
+
+        when:
+        def runner = GradleRunner.create()
+        runner.forwardOutput()
+        runner.withPluginClasspath()
+        runner.withArguments("licenseFormat")
+        runner.withProjectDir(projectDir)
+        def result = runner.build()
+
+        then:
+        result.output.contains("Task :updateLicenses UP-TO-DATE")
+        result.output.contains("Task :licenseFormat UP-TO-DATE")
+        result.task(":updateLicenses").outcome == TaskOutcome.UP_TO_DATE
+        result.task(":licenseFormat").outcome == TaskOutcome.UP_TO_DATE
+    }
+
+    def "supports custom source sets task"() {
+        given:
+        def projectDir = temporaryFolder.newFolder()
+        new File(projectDir, "settings.gradle") << ""
+        new File(projectDir, "build.gradle") << """
+            plugins {
+                id('org.cadixdev.licenser')
+                id('java')
+            }
+            sourceSets {
+                mySourceSet {}
+            }
+        """.stripIndent()
+
+        when:
+        def runner = GradleRunner.create()
+        runner.forwardOutput()
+        runner.withPluginClasspath()
+        runner.withArguments("licenseCheck")
+        runner.withProjectDir(projectDir)
+        def result = runner.build()
+
+        then:
+        result.task(":checkLicenseMySourceSet").outcome == TaskOutcome.NO_SOURCE
+    }
+
+    def "supports custom style"() {
+        given:
+        def projectDir = temporaryFolder.newFolder()
+        def sourcesDir = new File(projectDir, "sources")
+        sourcesDir.mkdirs()
+        new File(projectDir, "settings.gradle") << ""
+        new File(projectDir, "header.txt") << "Copyright header"
+        def sourceFile = new File(sourcesDir, "source.c") << "TEST"
+        new File(projectDir, "build.gradle") << """
+            plugins {
+                id('org.cadixdev.licenser')
+            }
+            
+            license {
+                header = project.file("header.txt")
+                style {
+                    c = 'BLOCK_COMMENT'
+                }
+                tasks {
+                    sources {
+                        files = project.files("sources")
+                        include("**/*.c")
+                    }
+                }
+            }
+        """.stripIndent()
+
+        when:
+        def runner = GradleRunner.create()
+        runner.forwardOutput()
+        runner.withPluginClasspath()
+        runner.withArguments("updateLicenses")
+        runner.withProjectDir(projectDir)
+        def result = runner.build()
+
+        then:
+        result.task(":updateLicenseCustomSources").outcome == TaskOutcome.SUCCESS
+        sourceFile.text == """\
+            /*
+             * Copyright header
+             */
+             
+            TEST
+            """.stripIndent()
+    }
+}

--- a/src/main/groovy/org/cadixdev/gradle/licenser/LicenseExtension.groovy
+++ b/src/main/groovy/org/cadixdev/gradle/licenser/LicenseExtension.groovy
@@ -176,7 +176,7 @@ class LicenseExtension extends LicenseProperties {
      * @param closure The closure to apply to the style
      */
     void style(@DelegatesTo(HeaderFormatRegistry) Closure closure) {
-        ConfigureUtil.configure(closure, style)
+        style.with(closure)
     }
 
     /**

--- a/src/test/groovy/org/cadixdev/gradle/licenser/LicenseExtensionTest.groovy
+++ b/src/test/groovy/org/cadixdev/gradle/licenser/LicenseExtensionTest.groovy
@@ -1,0 +1,46 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015, Minecrell <https://github.com/Minecrell>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.cadixdev.gradle.licenser
+
+
+import org.cadixdev.gradle.licenser.header.HeaderStyle
+import org.gradle.testfixtures.ProjectBuilder
+import spock.lang.Specification
+
+class LicenseExtensionTest extends Specification {
+    def "style closure registers header style"() {
+        given:
+        def project = ProjectBuilder.builder().build()
+        def licenseExtension = new LicenseExtension(project)
+
+        when:
+        licenseExtension.style {
+            ext = 'BLOCK_COMMENT'
+        }
+
+        then:
+        licenseExtension.style.getProperty("ext") == HeaderStyle.BLOCK_COMMENT.format
+    }
+}

--- a/src/test/groovy/org/cadixdev/gradle/licenser/LicenseExtensionTest.groovy
+++ b/src/test/groovy/org/cadixdev/gradle/licenser/LicenseExtensionTest.groovy
@@ -24,7 +24,6 @@
 
 package org.cadixdev.gradle.licenser
 
-
 import org.cadixdev.gradle.licenser.header.HeaderStyle
 import org.gradle.testfixtures.ProjectBuilder
 import spock.lang.Specification

--- a/src/test/groovy/org/cadixdev/gradle/licenser/LicenserPluginTest.groovy
+++ b/src/test/groovy/org/cadixdev/gradle/licenser/LicenserPluginTest.groovy
@@ -37,7 +37,7 @@ class LicenserPluginTest extends Specification {
 
         then:
         ["licenseCheck", "licenseFormat", "checkLicenses", "updateLicenses"].each {
-            project.tasks.findByName(it) != null
+            assert project.tasks.findByName(it) != null
         }
     }
 }

--- a/src/test/groovy/org/cadixdev/gradle/licenser/LicenserPluginTest.groovy
+++ b/src/test/groovy/org/cadixdev/gradle/licenser/LicenserPluginTest.groovy
@@ -1,0 +1,43 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015, Minecrell <https://github.com/Minecrell>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.cadixdev.gradle.licenser
+
+import org.gradle.testfixtures.ProjectBuilder
+import spock.lang.Specification
+
+class LicenserPluginTest extends Specification {
+    def "plugin registers tasks"() {
+        given:
+        def project = ProjectBuilder.builder().build()
+
+        when:
+        project.plugins.apply("org.cadixdev.licenser")
+
+        then:
+        ["licenseCheck", "licenseFormat", "checkLicenses", "updateLicenses"].each {
+            project.tasks.findByName(it) != null
+        }
+    }
+}

--- a/src/test/groovy/org/cadixdev/gradle/licenser/header/HeaderFormatRegistryTest.groovy
+++ b/src/test/groovy/org/cadixdev/gradle/licenser/header/HeaderFormatRegistryTest.groovy
@@ -1,0 +1,40 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015, Minecrell <https://github.com/Minecrell>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.cadixdev.gradle.licenser.header
+
+
+import spock.lang.Specification
+
+class HeaderFormatRegistryTest extends Specification {
+    def "registry contains default properties"() {
+        when:
+        def registry = new HeaderFormatRegistry()
+
+        then:
+        ["java", "js", "kt", "groovy", "yml", "xml", "html"].each {
+            registry.getProperty(it) != null
+        }
+    }
+}

--- a/src/test/groovy/org/cadixdev/gradle/licenser/header/HeaderFormatRegistryTest.groovy
+++ b/src/test/groovy/org/cadixdev/gradle/licenser/header/HeaderFormatRegistryTest.groovy
@@ -24,7 +24,6 @@
 
 package org.cadixdev.gradle.licenser.header
 
-
 import spock.lang.Specification
 
 class HeaderFormatRegistryTest extends Specification {
@@ -33,8 +32,6 @@ class HeaderFormatRegistryTest extends Specification {
         def registry = new HeaderFormatRegistry()
 
         then:
-        ["java", "js", "kt", "groovy", "yml", "xml", "html"].each {
-            registry.getProperty(it) != null
-        }
+        registry.keySet().containsAll(["java", "js", "kt", "groovy", "yml", "xml", "html"])
     }
 }

